infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/util/GatewayConfigmapLabels.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.workspace.infrastructure.kubernetes.util;
+
+import static org.eclipse.che.workspace.infrastructure.kubernetes.namespace.KubernetesObjectUtil.isLabeled;
+
+import com.google.common.base.Splitter;
+import io.fabric8.kubernetes.api.model.ConfigMap;
+import java.util.Map;
+import java.util.Map.Entry;
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+import org.eclipse.che.api.workspace.server.spi.InfrastructureException;
+
+/** Little utility bean helping with Gateway ConfigMaps labels. */
+@Singleton
+public class GatewayConfigmapLabels {
+
+  private final Map<String, String> labels;
+
+  @Inject
+  public GatewayConfigmapLabels(
+      @Named("che.infra.kubernetes.singlehost.gateway.configmap_labels") String labelsProperty)
+      throws InfrastructureException {
+    if (labelsProperty == null || labelsProperty.isEmpty()) {
+      throw new InfrastructureException(
+          "for gateway single-host, 'che.infra.kubernetes.singlehost.gateway.configmap_labels' property must be defined");
+    }
+    try {
+      this.labels = Splitter.on(",").trimResults().withKeyValueSeparator("=").split(labelsProperty);
+    } catch (IllegalArgumentException iae) {
+      throw new InfrastructureException(
```

Comment:
does it really InfrastructureException? I would say it's an invalid configuration.

Reply:
changed to `org.eclipse.che.inject.ConfigurationException`